planning/behavior_path_planner/src/planner_manager.cpp

```diff
@@ -378,6 +433,52 @@ BehaviorModuleOutput PlannerManager::getReferencePath(
   return output;
 }
 
+void PlannerManager::updateCandidateModules(
+  const std::vector<SceneModulePtr> & candidate_modules,
+  const SceneModulePtr & highest_priority_module)
+{
+  const auto exist = [](const auto & module_ptr, const auto & module_ptrs) {
+    const auto itr = std::find_if(
+      module_ptrs.begin(), module_ptrs.end(),
+      [&module_ptr](const auto & m) { return m->name() == module_ptr->name(); });
+
+    return itr != module_ptrs.end();
+  };
+
+  for (auto itr = candidate_module_ptrs_.begin(); itr != candidate_module_ptrs_.end();) {
+    if (!exist(*itr, candidate_modules)) {
+      deleteExpiredModules(*itr);
+      itr = candidate_module_ptrs_.erase(itr);
+      continue;
+    }
+
+    if (
+      (*itr)->name() == highest_priority_module->name() &&
+      !highest_priority_module->isWaitingApproval()) {
+      itr = candidate_module_ptrs_.erase(itr);
+      continue;
+    }
+
+    itr++;
+  }
```

Comment:
```suggestion
  const auto candidateToRemove = [&](auto & itr) {
    if (!exist(itr, candidate_modules)) {
      deleteExpiredModules(itr);
      return true;
    }

    return itr->name() == highest_priority_module->name() &&
           !highest_priority_module->isWaitingApproval();
  };

  candidate_module_ptrs_.erase(
    std::remove_if(candidate_module_ptrs_.begin(), candidate_module_ptrs_.end(), candidateToRemove),
    candidate_module_ptrs_.end());

```

In my understanding, the aim of this part is to remove the unused/invalid candidates. Therefore, I would suggest using remove_if instead of manually moving iterators.
What is Ota-san opinion on this.